internal/credit/postgres_connector/grant.go

```diff
@@ -24,6 +24,11 @@ func (c *PostgresConnector) CreateGrant(ctx context.Context, grantIn credit.Gran
 			return nil, err
 		}
 
+		// truncate effeectiveAt to start of next window
```

Comment:
Please add some comments detailing the issue why we need to ceil the timestamp here (this will not be obvious to others besides me and you).